web/src/app/ee/admin/groups/page.tsx

```diff
@@ -32,6 +34,24 @@ const Main = () => {
     error: usersError,
   } = useUsers();
 
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const isAdmin = currentUser?.role === UserRole.ADMIN;
+  useEffect(() => {
+    const fetchCurrentUser = async () => {
+      try {
+        const user = await getCurrentUser();
+        if (user) {
+          setCurrentUser(user);
+        } else {
+          console.error("Failed to fetch current user");
+        }
+      } catch (error) {
+        console.error("Error fetching current user:", error);
+      }
+    };
+    fetchCurrentUser();
+  }, []);
```

Comment:
Seeing a lot of these throughout the code - would definitely abstract away into a hook unless you're explicitly using `setCurrentUser` - even then, could be passed by the hook. Checkout `lib/hooks.ts` for some inspiration!